### PR TITLE
pyTEMlib to conda-forge

### PR DESCRIPTION
**Pull request by gduscher**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -30,13 +28,11 @@ build:
   # It is okay if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
   # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
   # noarch: python
+  number: 0
```
Comment (jaimergp) [inline recipes/example/meta.yaml:31]: Please revert the changes done to `example/meta.yaml`.

File: recipes/pyTEMlib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 20f262cbdc89602a0897bd6324b6ed9b313496f98b2310217e9cbaadc765e444
+
+build:
+  noarch: python
+  number: 0 
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (jaimergp) [inline recipes/pyTEMlib/meta.yaml:15]: Add the corresponding [`entry_points`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points) item here.

Comment (jaimergp) [reply recipes/pyTEMlib/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
  entry_points:
    - pyTEMlib = pyTEMlib:main
```

**Comment by gduscher** [comment]
updated entry point
I am really grateful to you Jamie to get us going, Thank you.

**Comment by jaimergp** [comment]
I can't see the changes @gduscher :(

We also need to revert the edits on the example yaml.

**Comment by gduscher** [comment]
Somehow the last line in the meta.yaml file in example was eaten.
Should be fixed now.

File: recipes/pyTEMlib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    - spglib
+    - scikit-image
+    - scikit-learn
+    - pyNSID
+    - sidpy
+    - SciFiReaders
+
+test:
+  imports:
+    - pyTEMlib
```
Comment (jaimergp) [inline recipes/pyTEMlib/meta.yaml:43]:
```suggestion
    - pyTEMlib
    - pyTEMlib.animation
    - pyTEMlib.atom_tools
    - pyTEMlib.config_dir
    - pyTEMlib.crystal_tools
    - pyTEMlib.diffraction_plot
    - pyTEMlib.dynamic_scattering
    - pyTEMlib.edges_db
    - pyTEMlib.eels_dialog
    - pyTEMlib.eels_dlg
    - pyTEMlib.eels_tools
    - pyTEMlib.file_tools
    - pyTEMlib.file_tools_qt
    - pyTEMlib.graph_tools
    - pyTEMlib.graph_viz
    - pyTEMlib.image_dialog
    - pyTEMlib.image_dlg
    - pyTEMlib.image_tools
    - pyTEMlib.info_dialog
    - pyTEMlib.info_dlg
    - pyTEMlib.interactive_eels
    - pyTEMlib.interactive_image
    - pyTEMlib.kinematic_scattering
    - pyTEMlib.microscope
    - pyTEMlib.peak_dialog
    - pyTEMlib.peak_dlg
    - pyTEMlib.probe_tools
    - pyTEMlib.sidpy_tools
    - pyTEMlib.simulation_tools
    - pyTEMlib.viz
```

File: recipes/pyTEMlib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    - tqdm
+    - ipyfilechooser
+    - plotly
+    - pandas
+    - spglib
+    - scikit-image
+    - scikit-learn
+    - pyNSID
+    - sidpy
+    - SciFiReaders
```
Comment (jaimergp) [inline recipes/pyTEMlib/meta.yaml:39]:
```suggestion
    - SciFiReaders
    - h5py >=3.4
```

**Comment by jaimergp** [comment]
There's an issue with the h5py dependency. If not specified, the solver will pick it up as a 2nd-order dep, all they way back to 3.1. However, those builds have an issue with hdf5 ABI compatibility. Forcing a more recent version causes a conflict with itk stuff on macos :/

**Comment by gduscher** [comment]
Finally, the package passes.

**Comment by jaimergp** [comment]
Thanks for all the work in making this pass!
